packages/next-urql/src/with-urql-client.ts

```diff
@@ -48,14 +48,14 @@ export function withUrqlClient(
 
       // eslint-disable-next-line react-hooks/rules-of-hooks
       const client = React.useMemo(() => {
-        if (urqlClient) {
+        if (urqlClient && forceUpdate[0] === 0) {
           return urqlClient;
         }
 
         if (!ssr || typeof window === 'undefined') {
           // We want to force the cache to hydrate, we do this by setting the isClient flag to true
           ssr = ssrExchange({ initialState: urqlServerState, isClient: true });
-        } else if (ssr && typeof window !== 'undefined') {
+        } else if (forceUpdate[0] === 0) {
```

Comment:
`ssr && typeof window !== 'undefined` is removed because that condition is automatically implied in the else if block due to demorgan's law